Fix invalid generation when using in-rule and not_in rule in map.keys.string

ucpr opened the pull request:
## Summary

I fixed invalid generation for Go and cc when using in-rule and not_in rule in map.keys.string

## Issue

Closes https://github.com/bufbuild/protoc-gen-validate/issues/785

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bufbuild/protoc-gen-validate?pullRequest=847) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bufbuild/protoc-gen-validate?pullRequest=847) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bufbuild/protoc-gen-validate?pullRequest=847) it.</sub>

elliotmjackson commented:
Thanks for the contribution @ucpr! Can you add a test case to the [harness](https://github.com/bufbuild/protoc-gen-validate/tree/main/tests/harness) so that we can ensure this doesn't regress?

ucpr commented:
Hi @elliotmjackson!

I've added some test cases to the PR. 
Adding the test case caused the C++ test to fail in the local environment. Therefore, I fixed it with Go in this PR.

ucpr commented:
Oh, test is failing. I will fix it.

ucpr commented:
@elliotmjackson I fixed tests

ucpr commented:
@elliotmjackson 
CI has passed. Please review at your convenience 👍🏼
